controllers/callhomeconfig_controller.go

```diff
@@ -139,41 +143,56 @@ func (r *CallHomeConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			fmt.Sprintf("unexpected error getting service account: %s", err),
 			ctrl.Result{})
 	}
-	if len(sa.Secrets) < 1 {
-		return retryLater(r, log, &chc, "serviceaccount",
-			namespacedName.Name, "does not have a token")
-	}
-	tokenSecretName := sa.Secrets[0]
-	var token corev1.Secret
-	if err := r.Get(ctx, types.NamespacedName{
+	var secretList corev1.SecretList
+	var kubeconfigSecret corev1.Secret
+	secretFound := false
+	if err := r.List(ctx, &secretList, &client.ListOptions{
 		Namespace: req.Namespace,
-		Name:      tokenSecretName.Name,
-	}, &token); err != nil {
-		if apierrors.IsNotFound(err) {
-			return retryLater(r, log, &chc, "token secret",
-				tokenSecretName.Name, "does not exist yet")
+	}); err != nil {
+		return retryLater(r, log, &chc, "secrets", req.Namespace, "cannot list secrets")
```

Comment:
It's a little weird to pass "secrets" and req.Namespace as 4th and 5th parameters. The retryLater() function will log those as the resource type and name that is experiencing a failure. Perhaps pass "namespace" instead of "secrets".

Reply:
done